internal/command/unlock.go

```diff
@@ -95,6 +100,17 @@ func (c *UnlockCommand) Run(args []string) int {
 
 	_, isLocal := stateMgr.(*statemgr.Filesystem)
 
+	if backendConfig != nil && (backendConfig.Type == "s3" || backendConfig.Type == "oss" || backendConfig.Type == "http") {
+		if optionalLocker, ok := stateMgr.(OptionalLocker); ok {
+			// Now we can safely call IsLockingEnabled() on optionalLocker
+			if !optionalLocker.IsLockingEnabled() {
+				c.Ui.Error("Locking is not configured for this backend")
```

Comment:
It could be configured, but still disabled, so I suggest slightly rewording the message:

```suggestion
				c.Ui.Error("Locking is disabled for this backend")
```